apps/api/src/index.ts

```diff
@@ -275,26 +275,28 @@ const mainServer = app.listen(PORT, "0.0.0.0", () => {
 // Start dedicated metrics server on separate port
 const metricsServer = startMetricsServer();
 
+let shuttingDown = false;
 // Graceful shutdown handler
-const gracefulShutdown = (signal: string) => {
+const gracefulShutdown = async (signal: string) => {
+  if (shuttingDown) return;
+  shuttingDown = true;
+
   console.log(
     `\n[${signal}] Received shutdown signal, closing servers gracefully...`,
   );
 
   // Close both servers
   mainServer.close(() => {
-    console.log("Main server closed");
-  });
+    console.log("[Shutdown] Main server closed");
 
-  metricsServer.close(() => {
-    console.log("Metrics server closed");
-  });
+    metricsServer.close(() => {
+      console.log("[Shutdown] Metrics server closed");
 
-  // Force exit after timeout if graceful shutdown fails
-  setTimeout(() => {
```

Comment:
This comment and use of setTimeout is kind of confusing since the only thing keeping the process running is waiting for the timeout, and it will always wait.  So it's not exactly force exiting, it's forcing the process to stay running. 
Instead of this we can just use `process.exit(0)` since the callbacks all execute sync.